token_dispenser_client/token_dispenser_client.py

```diff
@@ -0,0 +1,149 @@
+"""
+A module discovers and invoke Token Dispenser Service (TDS) lambda
+"""
+import logging
+import json
+from typing import List
+from botocore.exceptions import ClientError
+import boto3
+lambda_client = boto3.client('lambda')
+ssm = boto3.client('ssm')
+# ssm parameter store default path for TDS ARN
+_DEFAULT_SSM_PATH_: str = '/service/token-dispenser'
+logger = logging.getLogger(__name__)
+
+
+def get_parameter_by_name(name: str) -> str:
+    """
+
+    Args:
+        name: the ssm name point to TDS arn
+
+    Returns: tds arn as string
+    """
+    response = ssm.get_parameter(
+        Name=name
+    )
+    logger.debug('get_parameter_by_name with name: %s, response: %s', name, response)
+    if 'Parameter' not in response:
+        raise ValueError(f"Error: Parameter not found for name: {name}")
```

Comment:
Looks like boto3 does throw an exception on a key not found instead of an empty response - https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ssm/client/get_parameter.html

That's my bad for not pointing that behavior out earlier